File: widest-vertical-area-between-two-points-containing-no-points.cpp
// 1637

class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        int ans;
        
        vector <int> v(points.size(), 0);
        
        for (int i = 0; i < points.size(); i++) {
            v[i] = points[i][0];
        }
        
        sort(v.begin(), v.end());
        
        if (v.size() == 1) {
            return 0;
        }
        
        ans = v[1] - v[0];
        
        for (int i = 1; i < v.size() - 1; i++) {
            ans = max(ans, v[i + 1] - v[i]);
        }
        return ans;
    }
};

File: widest-vertical-area-between-two-points-containing-no-points.py
from collections import defaultdict


class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:

        a = defaultdict()

        for i in points:
            if a.get(i[0]) is None:
                a[i[0]] = 1

        if len(a) == 1:
            return 0

        a = list(a.keys())
        a.sort()
        ans = a[1]-a[0]

        for i in range(1, len(a) - 1):
            if a[i + 1] - a[i] > ans:
                ans = a[i + 1] - a[i]
        return ans
